node_shell.py

```diff
@@ -387,10 +851,271 @@ def do_exit(self, arg):  # pylint: disable=unused-argument
         """Exit the shell."""
         return True
 
+    graph_gen_parser = cmd2.Cmd2ArgumentParser()
+    graph_gen_parser.add_argument('--link-types', '-l', default='all')
+    graph_gen_parser.add_argument(
+        '--identifier',
+        default='uuid',
+    )
+    graph_gen_parser.add_argument('--ancestor-depth', type=int)
+    graph_gen_parser.add_argument('--descendant-depth', type=int)
+    graph_gen_parser.add_argument('--process-out', action='store_true')
+    graph_gen_parser.add_argument('--process-in', action='store_true')
+    graph_gen_parser.add_argument('--verbose', action='store_true')
+    graph_gen_parser.add_argument('--engine', '-e', default='dot')
+    graph_gen_parser.add_argument('--output-format', '-f', default='pdf')
+    graph_gen_parser.add_argument('--show', '-s', action='store_true')
+
+    @needs_new_node
+    @cmd2.with_argparser(graph_gen_parser)
+    def do_graph_generate(self, arg):
```

Comment:
Considering that we now have direct piping to verdi, maybe we should avoid reduplicating commands and trying to replicate the same interface? This will be hard to maintain in the future. And now (also with my PR) there is tab completion + you can use `{}` for the PK. I would thus be tempted to remove this, unless this can give additional interactiveness w.r.t. simply verdi